lua/cmp/core.lua

```diff
@@ -72,18 +72,30 @@ end
 ---@type table<number, cmp.Source>
 core.sources = {}
 
+---@type table<string, table<cmp.Source>>
+core.sources_by_name = {}
+
 ---@type cmp.Context
 core.context = context.new()
 
 ---Register source
 ---@param s cmp.Source
 core.register_source = function(s)
   core.sources[s.id] = s
+  if not vim.tbl_contains(core.sources_by_name, s.name) then
```

Comment:
You should use `if not core.sources_by_name[s.name] then` instead. Because it is not a `List` but `Dict`.